modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceResponse.java

```diff
@@ -168,6 +168,23 @@ public void start(NHttpServerConnection conn) throws IOException, HttpException
                 SourceContext.getRequest(conn).getRequest());
         
         sourceConfiguration.getHttpProcessor().process(response, conn.getContext());
+
+        //Since HTTP HEAD request doesn't contain message body, entity will be null when above HttpProcessor()
+        // process invoked. Hence content length is set to 0 inside the HTTP-Core. To avoid that content length of
+        // the backend response is set as the content length.
+        if (entity == null &&
+            PassThroughConstants.HTTP_HEAD.equalsIgnoreCase(request.getRequest().getRequestLine().getMethod())) {
+            if (response.getFirstHeader(PassThroughConstants.ORGINAL_CONTEN_LENGTH) == null && (response
+                    .getFirstHeader(HTTP.CONTENT_LEN).getValue().equals("0"))) {
```

Comment:
Isn't it required to do a null check here getFirstHeader(HTTP.CONTENT_LEN).getValue()